### ISSUE TITLE
Clients do not close properly

### ISSUE THREAD

**Issue by NightOnTheSun** [Ling2000/project-base-on-LWE-CS419-#3]
The client files both continue to run in the background after closing the GUI. They do not disconnect from the server when closed; killing them manually results in the server throwing 2 errors.

 ConnectionResetError: [WinError 10054] An existing connection was forcibly closed by the remote host 
 TypeError: 'function' object does not support item deletion

**Comment by Ling2000** [comment]
Yes, Closing the GUI does not mean closing the server. The only way to quit is to kill them all.

**Comment by symxmyz233** [comment]
It is **a minor bug**.
